entrypoints/apply-shim.js

```diff
@@ -36,18 +36,21 @@ class ApplyShimInterface {
       return;
     }
     this.customStyleInterface = window.ShadyCSS.CustomStyleInterface;
-    if (this.customStyleInterface) {
-      this.customStyleInterface['transformCallback'] = (style) => {
-        applyShim.transformCustomStyle(style);
-      };
-      this.customStyleInterface['validateCallback'] = () => {
-        requestAnimationFrame(() => {
-          if (this.customStyleInterface['enqueued']) {
-            this.flushCustomStyles();
-          }
-        });
-      }
+    if (!this.customStyleInterface) {
+      this.customStyleInterface = new CustomStyleInterface();
+      window.ShadyCSS.CustomStyleInterface = this.customStyleInterface;
     }
+    this.customStyleInterface['transformCallback'] = (style) => {
```

Comment:
Do these hooks not have to compose? If so, maybe there could be an add/remove hook?

Reply:
No, the ScopingShim's version of these hooks entirely supersede ApplyShim's version, so there is no need to compose them.